File: web/src/constants/bisindoLabels.ts
/**
 * Tipe Data generik untuk mendefinisikan kamus pelafalan dan teks murni huruf isyarat BISINDO.
 */
export interface BisindoSignLabel {
  id: number;
  label: string;
  isWord: boolean; // Menandakan apakah ini kata kerja (true) atau alfabet (false)
}

/**
 * Batas minimal probabilitas deteksi (confidence score).
 * Deteksi dengan skor di bawah ini akan difilter/diabaikan.
 */
export const BISINDO_CONFIDENCE_THRESHOLD = 0.25;

/**
 * Ukuran input model YOLOv8 dalam piksel (width = height).
 */
export const BISINDO_MODEL_INPUT_SIZE = 640;

/**
 * Tabel Lookup 48 Kelas Isyarat BISINDO (Alfabet & Kata).
 * Disusun berdasarkan urutan model YOLOv8 hasil training.
 */
export const BISINDO_LABELS: BisindoSignLabel[] = [
  { id: 0,  label: "A", isWord: false },
  { id: 1,  label: "B", isWord: false },
  { id: 2,  label: "C", isWord: false },
  { id: 3,  label: "D", isWord: false },
  { id: 4,  label: "E", isWord: false },
  { id: 5,  label: "F", isWord: false },
  { id: 6,  label: "G", isWord: false },
  { id: 7,  label: "H", isWord: false },
  { id: 8,  label: "I", isWord: false },
  { id: 9,  label: "J", isWord: false },
  { id: 10, label: "K", isWord: false },
  { id: 11, label: "L", isWord: false },
  { id: 12, label: "M", isWord: false },
  { id: 13, label: "N", isWord: false },
  { id: 14, label: "O", isWord: false },
  { id: 15, label: "P", isWord: false },
  { id: 16, label: "Q", isWord: false },
  { id: 17, label: "R", isWord: false },
  { id: 18, label: "S", isWord: false },
  { id: 19, label: "T", isWord: false },
  { id: 20, label: "U", isWord: false },
  { id: 21, label: "V", isWord: false },
  { id: 22, label: "W", isWord: false },
  { id: 23, label: "X", isWord: false },
  { id: 24, label: "Y", isWord: false },
  { id: 25, label: "Z", isWord: false },
];

File: mobile/lib/core/constants/bisindo_labels.dart
/// Model data untuk memetakan kelas output YOLOv8 BISINDO ke dalam bentuk 
/// label dan membedakan antara Alfabet dan Kata Kerja.
class BisindoSignLabel {
  final int id;
  final String label;
  final String indo;
  final bool isWord; // Menandakan apakah kelas ini adalah kata (true) atau alfabet (false)

  const BisindoSignLabel({
    required this.id,
    required this.label,
    required this.indo,
    required this.isWord,
  });
}

/// Batas minimal probabilitas deteksi (confidence score).
const double kBisindoConfidenceThreshold = 0.25;

/// Ukuran input model YOLOv8 Bisindo dalam piksel (width = height).
const int kBisindoModelInputSize = 640;

/// Tabel Konstanta 26 Kelas Alfabet BISINDO (A–Z).
const List<BisindoSignLabel> bisindoLabelsData = [
  BisindoSignLabel(id: 0, label: "A", indo: "A", isWord: false),
  BisindoSignLabel(id: 1, label: "B", indo: "B", isWord: false),
  BisindoSignLabel(id: 2, label: "C", indo: "C", isWord: false),
  BisindoSignLabel(id: 3, label: "D", indo: "D", isWord: false),
  BisindoSignLabel(id: 4, label: "E", indo: "E", isWord: false),
  BisindoSignLabel(id: 5, label: "F", indo: "F", isWord: false),
  BisindoSignLabel(id: 6, label: "G", indo: "G", isWord: false),
  BisindoSignLabel(id: 7, label: "H", indo: "H", isWord: false),
  BisindoSignLabel(id: 8, label: "I", indo: "I", isWord: false),
  BisindoSignLabel(id: 9, label: "J", indo: "J", isWord: false),
  BisindoSignLabel(id: 10, label: "K", indo: "K", isWord: false),
  BisindoSignLabel(id: 11, label: "L", indo: "L", isWord: false),
  BisindoSignLabel(id: 12, label: "M", indo: "M", isWord: false),
  BisindoSignLabel(id: 13, label: "N", indo: "N", isWord: false),
  BisindoSignLabel(id: 14, label: "O", indo: "O", isWord: false),
  BisindoSignLabel(id: 15, label: "P", indo: "P", isWord: false),
  BisindoSignLabel(id: 16, label: "Q", indo: "Q", isWord: false),
  BisindoSignLabel(id: 17, label: "R", indo: "R", isWord: false),
  BisindoSignLabel(id: 18, label: "S", indo: "S", isWord: false),
  BisindoSignLabel(id: 19, label: "T", indo: "T", isWord: false),
  BisindoSignLabel(id: 20, label: "U", indo: "U", isWord: false),
  BisindoSignLabel(id: 21, label: "V", indo: "V", isWord: false),
  BisindoSignLabel(id: 22, label: "W", indo: "W", isWord: false),
  BisindoSignLabel(id: 23, label: "X", indo: "X", isWord: false),
  BisindoSignLabel(id: 24, label: "Y", indo: "Y", isWord: false),
  BisindoSignLabel(id: 25, label: "Z", indo: "Z", isWord: false),
];
